rust/hook-api/src/handlers/app.rs

```diff
@@ -5,17 +5,32 @@ use hook_common::pgqueue::PgQueue;
 
 use super::webhook;
 
-pub fn add_routes(router: Router, pg_pool: PgQueue, max_body_size: usize) -> Router {
-    router
+pub fn add_routes(
+    router: Router,
+    pg_pool: PgQueue,
+    hog_mode: bool,
+    max_body_size: usize,
+) -> Router {
+    let router = router
         .route("/", routing::get(index))
         .route("/_readiness", routing::get(index))
-        .route("/_liveness", routing::get(index)) // No async loop for now, just check axum health
-        .route(
+        .route("/_liveness", routing::get(index)); // No async loop for now, just check axum health
+
+    if hog_mode {
+        router.route(
+            "/hoghook",
```

Comment:
Whats the thinking behind "hog mode"? Why not just use the same webhook service? If you think its worth the extra overhead and complexity then happy to move forward, just wondered if this makes it all a bit more complex to manage. Like is there something specifically we are worried about here?

Reply:
After looking through the whole PR I get it a little more. I just wonder if it wouldn't be simpler to just always support hog calls, that way there would be way less boilerplate but I defer to you on that.

Reply:
Some of my thinking:

* The hooks service was written so that it would be easy to spin up additional shards, either for scale or for making data model changes (or risky changes in general). The idea being that all data in the queue is relatively short lived, and so you'd spin up a new version, switch new traffic to it, drain the old queue, and delete the old version.
* Since the Hog code/use is new, I figured we'd want to run a hooks service just for Hog while we test it out (and avoid hurting real prod traffic), so this is sort of leaning into that.
* The prod hooks service requires parameters Hog doesn't care about (plugin related) and uses those to push things to the `app_metrics` table. We don't want that for Hog, and adding the runtime code to mark each job as Hog vs not-Hog felt like more work than a few high level `if`s.
* With any luck, those Hog `if`s are a clear demarkation of what code to drop in the future when Hog takes over, rather than supporting both use cases via more complex query/janitor logic.

All that said, it's not too strong of an opinion, it's just sort of the route I took. For what it's worth, I think the first bullet is probably most important, and is why I feel like there's little downside to having "Hog mode."

Reply:
No cool im sold - lets carry on this way. Thanks for the context 🙌